File: intellicart_frontend/lib/data/dtos/user_dto.dart
// lib/data/dtos/user_dto.dart

class UserDto {
  final String id;
  final String email;
  final String name;
  final String role;
  
  UserDto({
    required this.id,
    required this.email,
    required this.name,
    required this.role,
  });
  
  factory UserDto.fromJson(Map<String, dynamic> json) {
    return UserDto(
      id: json['id'] ?? '',
      email: json['email'] ?? '',
      name: json['name'] ?? '',
      role: json['role'] ?? 'buyer',
    );
  }
  
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'email': email,
      'name': name,
      'role': role,
    };
  }
}

File: intellicart_backend/src/models/UserDTO.ts
/**
 * User Data Transfer Object (DTO) Module
 * 
 * This module defines the data transfer objects for user-related operations.
 * These interfaces provide type safety and clear structure for user data
 * across the application.
 * 
 * @module UserDTO
 * @description DTOs for user data structures
 * @author Intellicart Team
 * @version 1.0.0
 */

// Interface for a User entity
export interface User {
  id: number;
  name: string;
  email: string;
  password?: string; // Password is optional since it may not be sent in responses
  role?: string; // 'buyer' or 'seller'
  age?: number;
  createdAt?: string;
}

// Interface for creating a new user (request)
export interface CreateUserRequest {
  name: string;
  email: string;
  password: string;
  age?: number;
  role?: string;
}

// Interface for updating a user (request)
export interface UpdateUserRequest {
  name?: string;
  email?: string;
  age?: number;
  role?: string;
}

// Interface for user authentication response
export interface AuthenticatedUserResponse {
  user: User;
  token: string;
}

// Interface for login request
export interface LoginRequest {
  email: string;
  password: string;
}

// Interface for registration request
export interface RegisterRequest extends CreateUserRequest {
  role?: string;
}